### PR TITLE
feat: 1155 name/symbol, gas optimizations, approveTransfer w/operator

### PR DESCRIPTION
**Pull request by jtgi**

### CODE REVIEW

**Comment by wwhchung** [comment]
Changes include the following:
1. Gas optimizations
2. Allow extensions to override royalties
3. Modify IERC721CreatorExtensionApproveTransfer.approveTransfer to include the operator
4. Add name/symbol to ERC1155
5. Add version number variable

Note that you can write an IERC721CreatorExtensionApproveTransfer extension that is compitable with older contracts as long as:
 - you return true for supportsInterface for the interfaceId 0x99cdaa22, and
 - you implement approveTransfer WITHOUT operator

**Comment by wwhchung** [comment]
The changes to IERC721CreatorExtensionApproveTransfer will allow an extension  to support the OpenSea Operator Filter and any other similar filters.

File: contracts/ERC721Creator.sol [hunk 1]
```diff
@@ -152,28 +154,28 @@ contract ERC721Creator is AdminControl, ERC721, ERC721CreatorCore {
      */
     function mintBaseBatch(address to, uint16 count) public virtual override nonReentrant adminRequired returns(uint256[] memory tokenIds) {
         tokenIds = new uint256[](count);
-        for (uint16 i = 0; i < count; i++) {
+        for (uint16 i = 0; i < count;) {
             tokenIds[i] = _mintBase(to, "");
+            unchecked { ++i; }
         }
-        return tokenIds;
```
Comment (jtgi) [inline contracts/ERC721Creator.sol:158]: why is this compiling?

Comment (jtgi) [reply contracts/ERC721Creator.sol:158]: nvm https://docs.soliditylang.org/en/v0.8.17/contracts.html#return-variables

File: contracts/core/CreatorCore.sol [hunk 2]
```diff
@@ -269,42 +273,60 @@ abstract contract CreatorCore is ReentrancyGuard, ICreatorCore, ERC165 {
...
+            royalties = _extensionRoyalty[address(0)];
+        }
+        
+        if (royalties.length > 0) {
+            receivers = new address payable[](royalties.length);
+            bps = new uint256[](royalties.length);
+            for (uint256 i = 0; i < royalties.length;) {
+                receivers[i] = royalties[i].receiver;
+                bps[i] = royalties[i].bps;
+                unchecked { i++; }
```
Comment (jtgi) [inline contracts/core/CreatorCore.sol:301]: ++i;

File: contracts/core/CreatorCore.sol [hunk 3]
```diff
@@ -269,42 +273,60 @@ abstract contract CreatorCore is ReentrancyGuard, ICreatorCore, ERC165 {
...
+    function _getRoyalties(uint256 tokenId) view internal returns (address payable[] memory receivers, uint256[] memory bps) {
+
+        // Get token level royalties
+        RoyaltyConfig[] memory royalties = _tokenRoyalty[tokenId];
+        if (royalties.length == 0) {
+            // Get extension specific royalties
+            address extension = _tokensExtension[tokenId];
+            if (extension != address(0)) {
+                if (ERC165Checker.supportsInterface(extension, type(ICreatorExtensionRoyalties).interfaceId)) {
+                    (receivers, bps) = ICreatorExtensionRoyalties(extension).getRoyalties(address(this), tokenId);
```
Comment (jtgi) [inline contracts/core/CreatorCore.sol:283]: value in sanity check here? require tbps < 10000 && receivers.length === bps.length

Comment (wwhchung) [reply contracts/core/CreatorCore.sol:283]: Don't know if I actually care, it's a read function.  This should be caught by anyone reading this data.

File: contracts/core/CreatorCore.sol (hunk 2 above)
Comment (wwhchung) [reply contracts/core/CreatorCore.sol]: I just fixed this. :)

**Pull request merged**